### ISSUE TITLE
Web based Cross-Platform GUI

### ISSUE THREAD

**Issue by timgrossmann** [InstaPy/InstaPy#651]
> This is a key issue!
[![Bountysource](https://api.bountysource.com/badge/issue?issue_id=49290853)](https://www.bountysource.com/issues/49290853-web-based-cross-platform-gui?utm_source=49290853&utm_medium=shield&utm_campaign=ISSUE_BADGE)

Create a web based GUI that acts similar to the [Windows GUI](https://github.com/Nemixalone/GUI-tool-for-InstaPy-script) but also works on Linux and MacOS Devices. The Webpage could be bundled into an Electron App with [Zeit's pkg module](https://github.com/zeit/pkg). It would be best if the result of this would be an executable file just like an .exe file on windows.

Take a look at the [Windows GUI](https://github.com/Nemixalone/GUI-tool-for-InstaPy-script) for the "Setup help" that is displayed on the bottom corner to indicate a missing chrome driver or not installed dependencies.

Also make sure to take a look at tools like:
- [https://pushgram.co](https://pushgram.co/wp-content/uploads/2017/03/devices_03.png)
- [Instagres](http://www.thomadaneau.com/wp-content/uploads/2015/03/Screen-Shot-2015-03-19-at-11.56.41-AM.png)
- [Massplanner](https://www.ecosia.org/images?q=massplanner)

Also make sure to take a look at how the setup of the Quickstart file is done there. 

DoD:
- I can setup my Quickstart file solely with the GUI
- Make it pretty (at least try to)
- Make it self-explaining
- It is easily extensible with more features
- It has tooltips that quickly explain what each feature does (maybe just the readme description or something)

This is open for discussion, any comments/improvements or **additional bounties** are always helpful!

**Comment by gitpatrickhub** [comment]
@timgrossmann It's a great idea. My only thought is that by making this GUI it will open up the tool to greater use and therefore, increase the likelihood for IG to try and prevent it's use.

**Comment by KaiHatchman** [comment]
Sounds like a really good idea

**Comment by timgrossmann** [comment]
@gitpatrickhub Yes, that is definitely possible. But I don't think investing all the hours for the support of the less computerphile people. This might lead to a quick and large increase in users which could therefore lead to some more funds to cover all the bugs with bounties.

The idea also is to create a nice landing page that invites the user to simply download the mentioned executable file (the whole folder with that in it).

Given the instructions, the user can then "simply" follow them to get it working without a lot of programming knowledge.

I know there are quite a lot of bugs right now, but once the overall features and setup is there, we can tackle them and make it less error prune. There is just too much change right now to only focus on the bugs.

Once everything is in place, V_1 will be next next goal. Restructuring the Project and making sure people can either simply get it from pip (which will again be only for the computer guys) or download the whole tool ready to use from a webpage

The very best case would be to slowly make all the services out there obsolete 😉

**Comment by timgrossmann** [comment]
@ahmadudin Any updates on this? 😉

**Comment by ahmadudin** [comment]
@timgrossmann yeah, scripting and UI part is done. Currently working on script execution part. I'm wondering, is it needed to show execution log in the UI ?

**Comment by timgrossmann** [comment]
@ahmadudin Would be a really nice to have :wink:

**Comment by ahmadudin** [comment]
@timgrossmann so the early functional app is up on my [repo](https://github.com/ahmadudin/electron-instaPy-GUI). Haven't upload any releases yet. So far it works good on windows, found some problem on linux (still working on it). I can't directly test it on mac since i don't have access to the device. Hope you can give it a try and give some feedback.

Sorry it takes too long to build this. Recently i have to work on my final year project during the day, so i can only do this at night.

**Comment by timgrossmann** [comment]
@ahmadudin Awesome! I hope I can find some time to test this tonigh.
Will get back to you, thank you for building this!

**Comment by timgrossmann** [comment]
@ahmadudin Hey, just tested the GUI and I really have to say, you did an incredibly awesome job!

Everything major works. I found some small problems which I'll list here. They are really minor though.
Thank you so much for creating this, it's awesome. I'll link it in the Readme.
The only thing that is left is to create an exe that people really easily can use.

Here are the things I noticed (sadly couldn't test all the features):
- If I put the chromedriver in assets, the GUI won't update saying it's there, only after restart
- The stop function of the Terminal does not work / stops the script but won't complete the stopping
- Couldn't test having no dependencies installed (are there some information if nothing is installed?)

I hope the rest is working. If people really start to use it we'll see if everything else is working as well.

Thank you again for this giant contribution in making InstaPy waaaaaaay better!
(I'll raise the bounty a little bit (currently seems to not work with PayPal, I'll try again tomorrow) to give an upfront investment for the bug fixes and minor adjustments you will probably have to make in the next days.)

Could you create an executable as soon as possible, please?

**Comment by ahmadudin** [comment]
@timgrossmann thank you for testing it! 

- The listed problems seem similiar to the problem that occur on linux, while in windows all these problems are not exist.
- File check is always scanning for a change before, it seems not working after i refactor the code. thank you for pointing it out.

There is some more feature that i want to add, but i will rush to finish basic function on all platform first. I will release executable as soon as the ptoblem is solved!

**Comment by timgrossmann** [comment]
@ahmadudin Sounds awesome! Thank you.

In combination with this new incredible GUI, we now only have to come up with a plan on how to make the setup as easy as clicking a Button so we can then distribute InstaPy through a single Landing page download button and people really easily can set everything up and start with the GUI and tool

**Comment by ahmadudin** [comment]
@timgrossmann I have released the gui installers, and i think thats it for now. I'll slow down the development for a while, for i have to fully focused working on my thesis. I'd still fixing minor bugs, but not major development.

**Comment by timgrossmann** [comment]
@ahmadudin Sure! Thank you very much!

Somehow the GitHub login for Bountysource is broken at the moment so I can't approve the dev of this issue... I hope it'll be back up soon, I'll add some on top of the bounty and approve you as the dev ASAP

**Comment by ahmadudin** [comment]
@timgrossmann Yeah, it seems something not right is going on bountysource. Last time i requested cashout from bountysource, it needs more than a month to be processed.

**Comment by ahmadudin** [comment]
@timgrossmann or is it possible for you to get refund from bountysource and send it directly? just to speed things up

**Comment by timgrossmann** [comment]
@ahmadudin I will check that...

Edit: Problem is, I can't login at all which means there is no refund...

I can send over the first half through PayPa ($50)l if you want to, once Bountysource is up again, I can send you the rest ($75).

**Comment by ahmadudin** [comment]
@timgrossmann i think its better that way, thank you tim. My paypal account ahmed.shalahuddin@gmail.com

**Comment by timgrossmann** [comment]
@ahmadudin Did you already get it?

**Comment by ahmadudin** [comment]
@timgrossmann Yes, thanks!

**Comment by timgrossmann** [comment]
@ahmadudin Just found a way to log into BountySource 😉 
Thank you again for this great tool!

**Comment by Coriou** [comment]
My two cents here.

Wouldn't it make sense to first make an executable (ie a single portable binary) of InstaPy in order to achieve this nicely ? Just a thought here, but I think it would make the work on the UI a lot simpler if InstaPy had a packaged version exposing just the CLI API the engine running the UI would interact with. Would greatly simplify installation process. Plus, it would offer developers who don't want the UI but still care about easy deployment, a really convenient way to use InstaPy, and integrate it with they existing toolchain (already possible today, a binary would just be easier).

Maybe I'm not seeing an obvious no-no about this, but right now it seems like the first step. Thoughts ?
